c/driver/postgresql/postgres_copy_reader.h

```diff
@@ -635,6 +635,19 @@ static inline ArrowErrorCode MakeCopyFieldReader(const PostgresType& pg_type,
         default:
           return ErrorCantConvert(error, pg_type, schema_view);
       }
+
+    case NANOARROW_TYPE_TIMESTAMP:
+      switch (pg_type.type_id()) {
+        case PostgresTypeId::kTimestamp: {
+          // 2000-01-01 00:00:00.000000 in microseconds
```

Comment:
Is this enforced on the arrow side? I though Postgres used the unix epoch?

Reply:
Other way around: Arrow uses the Unix epoch, but Postgres apparently uses 2000/1/1 as its epoch.

Reply:
Huh that's interesting. Strange that client queries don't work that way:

```psql
postgres=# SELECT 'epoch'::timestamp;
      timestamp      
---------------------
 1970-01-01 00:00:00
(1 row)
```

Something to watch out for in the future

Reply:
Yes, presumably the clients convert the value (like we do here), but the value on the wire is different (I extracted this constant by looking at it in the debugger, and if you take current HEAD you'll get back a binary array with this value encoded in it!)